### PR TITLE
Add --no-daemonize option to synctl

### LINKED ISSUE #4130: synctl should have a `--no-daemonize` option
Sometimes it's useful to be able to run synapse in the foreground for debugging. You can do this by running `python -m synapse.app.homeserver` directly, but that means you don't get quite the same environment as synctl gives you, and in any case it's opaque for users.

We should give synctl a `--no-daemonize` option, which makes it run synapse without the `--daemonize` option. 

(It would be an error to try to set `--no-daemonize` and `--all-processes` at the same time)

### PR DESCRIPTION
**Pull request by SohamG**
Fix #4130 

- Add parser argument "--no-daemonize"

Signed-off-by: sohamg <sohamg2@gmail.com>

### Pull Request Checklist

* [ ✅] Pull request is based on the develop branch
* [ ✅] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [✅ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

**Comment by erikjohnston** [comment]
This looks good, thanks! The only issue I can see is that this will not do the right thing if the daemonize option has been set in the config file. I think this is probably fine for now, though it would be great if you could either a) note in the argument description it only works if `daemonize` is not set in the config, or b) change the `--daemonize` config option in synapse to support both `--daemonize` and `--daemonize=no` so that we can then use that form for the `--no-daemonize` case

File: synctl [hunk 1]
```diff
@@ -69,10 +69,14 @@ def abort(message, colour=RED, stream=sys.stderr):
     sys.exit(1)
 
 
-def start(configfile):
+def start(configfile, daemonize = True):
```
Comment (erikjohnston) [inline synctl]: PEP8 is failing due to:

```
[2019-06-10T12:39:25Z] synctl:72:32: E251 unexpected spaces around keyword / parameter equals
```

Comment (SohamG) [reply synctl]: Fixed in 0afcbc6

**Comment by SohamG** [comment]
Hey, thanks for the review. I fixed the pep8 errors, and implemented option a) in your message for the time being. I'd like to implement option b too, it might take me some time though. Should option b be for another PR?

**Comment by erikjohnston** [comment]
Your last commit looks to have reintroduced the pep8 error. Once that's fixed I'm happy to merge this PR if you don't think you'll have time in the next couple of days to do option b.

**Pull request merged**